include/dmlc/optional.h

```diff
@@ -107,6 +108,8 @@ class optional {
   }
   /*! \brief whether this object is holding a value */
   explicit operator bool() const { return !is_none; }
+  /*! \brief alternate, more-explicit check of object value */
+  const bool IsNone() const { return is_none; }
```

Comment:
I've dropped IsNone() and added the C++17 standard's has_value(), which satisfies my desire for a more-explicit call.